session/pingpong/hermes_promise_settler.go

```diff
@@ -721,24 +722,22 @@ func (aps *hermesPromiseSettler) generateAgreementID() *big.Int {
 	return new(big.Int).SetBytes(agreementID)
 }
 
-func (aps *hermesPromiseSettler) getWithdrawalAmountViaHermes(chainID int64, hermesID, identity common.Address) (*big.Int, error) {
+func (aps *hermesPromiseSettler) getHermesData(chainID int64, hermesID, identity common.Address) (ConsumerData, error) {
```

Comment:
Structure `ConsumerData` is quite large. Can't we operate with pointer to it?

Reply:
Overoptimizing the very rare call, but we can. Done.